packages/core/src/browser/shell/tab-bars.ts

```diff
@@ -413,18 +413,6 @@ export class TabBarRenderer extends TabBar.Renderer {
         if (this.contextMenuRenderer && this.contextMenuPath && event.currentTarget instanceof HTMLElement) {
             event.stopPropagation();
             event.preventDefault();
-
-            if (this.tabBar) {
```

Comment:
To remove any code, one should always check first why it was added and then make sure that it still works. 'How to test' section of a PR should have such tests.

Reply:
From what I understand, the removed code was more like of a workaround for context menus to work. Funny that we open *context* menus without context too, commands have to look up in some global state to get information about this "pseudo"-context. Should something be changed regarding the way context is propagated? It would help 1000x to be able to send the tab-title that originated the context menu spawn.

Or we can add more state, like "contextMenuTargetTitle" next to the "currentTitle" property.

WDYT?

Reply:
> and then make sure that it still works.

note: In this case, it seems we don't want this code to work anymore, if we are to align with VS Code, then we should not automatically activate widgets when right click on their titles. Do you know if the removed code did more than that?

Reply:
would not it break context menus? they expect that current (last focused) widget to be correct, otherwise you will see menu items for other menu.

Reply:
> It would help 1000x to be able to send the tab-title that originated the context menu spawn.

I think it is too late for such change, there are many commands which don't expect it in theia repo and downstream projects, we will break all of them.

Reply:
> we will break all of them.

We're not 1.0 yet, this is the time ;)

It's not like I'm just talking about renaming a method too, this would be a change that makes sense IMO...

Meanwhile, what would you recommend? If we set this `currentTitle` thing, the widget pops up.

We can try to dance around this, so we may try different things.

Reply:
> We're not 1.0 yet, this is the time ;)

we should not break things which are used a lot regardless of 1.0. In this case it was a deliberate decision to base on the current widget to support different contexts with the same code, not only context menus.

yes, let's work it around somehow, btw one can already pass args to the context menu rendered as we do it in the toolbar widget, so maybe don't remove code, but pass args and change only some commands to respect it?

Reply:
> one can already pass args to the context menu rendered as we do it in the toolbar widget

Thanks for the pointer, that's perfect! We should've used that from the beginning I think.

> so maybe don't remove code, but pass args and change only some commands to respect it?

The issue that is being fixed here are editors popping up on right click. This was indirectly caused by the removed code. I think commands should really expect to find some context, right now it seems we send the `MouseEvent` by default, we'll see what we can do from here.